src/lua/LuaPiGui.cpp

```diff
@@ -1103,13 +1106,15 @@ static int l_pigui_get_axisbinding(lua_State *l)
 	const auto &joysticks = Input::GetJoysticks();
 	InputBindings::JoyAxis binding = {};
 
-	for (const auto &js : joysticks) {
-		const auto &axes = js.second.axes;
+	for (unsigned id = 0; id < joysticks.size(); ++id) {
+		const auto &axes = joysticks[id].axes;
 		for (size_t axis_num = 0; axis_num < axes.size(); axis_num++) {
 			float val = axes[axis_num].value;
-			if (std::abs(val) > 0.25) {
+			// we will cut off the possible joystick noise, as well as the fact that
+			// the trigger in a relaxed position means -1.0
+			if (std::abs(val) > 0.25 && std::abs(val) < 0.9) {
```

Comment:
I really don't like this hard-coded hack to avoid triggers constantly binding. I'd much prefer if the input system had some tracking for "at-rest" axes and only evaluate ones that have changed their values during this frame. That being said, a change of that size is out of scope for this PR, so I'd appreciate it if you'd add a `TODO:` to this comment for later work.

Reply:
@Web-eWorks There is also a problem with "value jitter"